File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.Arrays;

class Solution {
	private static final int MIN = Integer.MIN_VALUE;
	
	public int solution(int[][] routes) {
		int i;
		int len;
		int cnt;
		int camera;

		Arrays.sort(routes, (o1, o2) -> o1[1] - o2[1]);
		len = routes.length;
		camera = MIN;
		cnt = 0;
		for (i = 0; i < len; i++) {
			if (camera < routes[i][0]) {
				camera = routes[i][1];
				cnt++;
			}
		}
		return cnt;
	}
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    int i;
    int cnt;
    int size;
    int last;

    sort(routes.begin(), routes.end(), [](auto& a, auto& b) {
        return a[1] < b[1];
    });
    cnt = 1;
    last = routes[0][1];
    size = routes.size();
    for (i = 1; i < size; i++) {
        if (routes[i][0] > last) {
            cnt++;
            last = routes[i][1];
        }
    }
    return cnt;
}